plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java

```diff
@@ -176,6 +187,82 @@ public void testEnforcedCooldownPeriod() throws IOException {
     }
 
+    public void testRequestStats() throws Exception {
+        final String repository = createRepository(randomName());
+        final String index = "index-no-merges";
+        createIndex(index, Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .build());
+
+        final long nbDocs = randomLongBetween(100, 1000);
+        try (BackgroundIndexer indexer = new BackgroundIndexer(index, "_doc", client(), (int) nbDocs)) {
+            waitForDocs(nbDocs, indexer);
+        }
+
+        flushAndRefresh(index);
+        ForceMergeResponse forceMerge = client().admin().indices().prepareForceMerge(index).setFlush(true).setMaxNumSegments(1).get();
+        assertThat(forceMerge.getSuccessfulShards(), equalTo(1));
+        assertHitCount(client().prepareSearch(index).setSize(0).setTrackTotalHits(true).get(), nbDocs);
+
+        final String snapshot = "snapshot";
+        assertSuccessfulSnapshot(client().admin().cluster().prepareCreateSnapshot(repository, snapshot)
+            .setWaitForCompletion(true).setIndices(index));
+
+        assertAcked(client().admin().indices().prepareDelete(index));
+
+        assertSuccessfulRestore(client().admin().cluster().prepareRestoreSnapshot(repository, snapshot).setWaitForCompletion(true));
+        ensureGreen(index);
+        assertHitCount(client().prepareSearch(index).setSize(0).setTrackTotalHits(true).get(), nbDocs);
+
+        assertAcked(client().admin().cluster().prepareDeleteSnapshot(repository, snapshot).get());
+
+        final RepositoryStats repositoryStats = StreamSupport.stream(
+            internalCluster().getInstances(RepositoriesService.class).spliterator(), false)
+            .map(repositoriesService -> {
+                try {
+                    return repositoriesService.repository(repository);
+                } catch (RepositoryMissingException e) {
+                    return null;
+                }
+            })
+            .filter(r -> r != null)
+            .map(r -> r.stats())
+            .reduce((s1, s2) -> s1.combine(s2))
+            .get();
+        final long sdkGetCalls = repositoryStats.requestCounts.get("GET");
+        final long sdkListCalls = repositoryStats.requestCounts.get("LIST");
+
+        final long getCalls = handlers.values().stream()
+            .mapToLong(h -> {
+                if (h instanceof S3HttpHandler) {
+                    return ((S3HttpHandler) h).getCalls.get();
+                } else if (h instanceof S3ErroneousHttpHandler) {
+                    return ((S3ErroneousHttpHandler) h).getCalls.get();
+                } else {
+                    return 0L;
+                }
+            })
+            .sum();
+        final long listCalls = handlers.values().stream().filter(h -> h instanceof S3HttpHandler)
```

Comment:
You probably don't want this filter because it prevents you from seeing the erroneous handlers below in the mapping.